core-java-modules/core-java-io-2/src/test/java/com/baeldung/filenotfoundexception/FileNotFoundExceptionUnitTest.java

```diff
@@ -1,58 +1,115 @@
 package com.baeldung.filenotfoundexception;
 
-import org.junit.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.*;
-
 public class FileNotFoundExceptionUnitTest {
 
     private static final Logger LOG = LoggerFactory.getLogger(FileNotFoundExceptionUnitTest.class);
 
-    private String fileName = Double.toString(Math.random());
+    @TempDir
+    private Path tempDir;
 
-    @Test(expected = BusinessException.class)
-    public void raiseBusinessSpecificException() throws IOException {
-        try {
-            readFailingFile();
-        } catch (FileNotFoundException ex) {
-            throw new BusinessException("BusinessException: necessary file was not present.");
+    @Test
+    void whenFileDoesNotExist_thenFileNotFoundException() {
+        File notExistFile = tempDir.resolve("dummy.notExist")
+            .toFile();
+        assertThrows(FileNotFoundException.class, () -> {FileReader fileReader = new FileReader(notExistFile);});
+    }
+
+    @Test
+    void whenFileExistButInaccessible_thenFileNotFoundException() throws IOException {
+        File readOnlyFile = tempDir.resolve("dummy.ReadOnly")
+            .toFile();
+        // first creating a read-only file
+        try (PrintWriter printWriter = new PrintWriter(new FileWriter(readOnlyFile))) {
+            printWriter.println("Hi there!");
+            readOnlyFile.setReadOnly();
+        } catch (Exception e) {
+            throw e;
         }
```

Comment:
I was thinking more like this, we don't have a reason to catch it explicitly at all:
```suggestion
        }
```